### PR TITLE
Added combined dataset

### PR DESCRIPTION
**Pull request by szympie**

### CODE REVIEW

**Comment by myychal** [comment]
Doesn't this https://pytorch.org/docs/stable/data.html#torch.utils.data.TensorDataset give the same result?

**Comment by szympie** [comment]
It does I wasn't aware of that. Weird name for such class.

File: brats/data/datasets.py [hunk 1]
```diff
@@ -21,14 +22,31 @@ def __init__(self, root: str, transform=None):
...
+
+
+class CombinedDataset(data.Dataset):
+    """
+    Takes multiple datasets of the same length and combines them.
+    On `__getitem__(n)` it returns a tuple containing nth element of each dataset.
+    """
+
+    def __init__(self, *datasets: data.Dataset):
+        assert all(len(dataset) == len(datasets[0]) for dataset in datasets), "Length of both datasets must match"
```
Comment (khrynczenko) [inline brats/data/datasets.py]: Just a neatpick `all(len(dataset) == len(datasets[0]) for dataset in datasets[1:])`. Also
"Length of **all** datasets must be the same".

File: brats/data/datasets.py [hunk 2]
```diff
@@ -21,14 +22,31 @@ def __init__(self, root: str, transform=None):
         self._files = list(os.scandir(root))
         self._transform = transform
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._files)
 
-    def __getitem__(self, item):
-        scan = nib.load(self._files[item].path)
+    def __getitem__(self, idx: int) -> typing.Any:
```
Comment (khrynczenko) [inline brats/data/datasets.py]: Generally I see that typing is the one module u import classes from directly. Its good anyway.

Comment (szympie) [reply brats/data/datasets.py]: Can you elaborate? I'm not sure if I understand it correctly 😅

File: brats/data/datasets.py (hunk 1 above)
Comment (szympie) [reply brats/data/datasets.py]: Yeah, good catch 😄

**Pull request merged**